### PR TITLE
force=True option for drop_table and drop_dir

### PR DESCRIPTION
**Pull request by aaron-siegel**
Adds force=True option to drop_table and drop_dir. Also renames rm_dir to drop_dir.

### CODE REVIEW

**Comment by aaron-siegel** [comment]
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/217)

File: docs/release/api-cheat-sheet.md [hunk 1]
```diff
@@ -17,22 +17,24 @@ import pixeltable.functions as pxtf
...
-| Create a directory         | [`pxt.create_dir`][pixeltable.create_dir]('experiments')              |
-| Rename or move a directory | [`pxt.move`][pixeltable.move]('experiments', 'project_x.experiments') |
-| Delete a directory         | f = [`pxt.rm_dir`][pixeltable.rm_dir]('experiments')                  |
-| List directories           | [`pxt.list_dirs`][pixeltable.list_dirs]('project_x')                  |
+| Task                                    | Code                                                                  |
+|-----------------------------------------|-----------------------------------------------------------------------|
+| Create a directory                      | [`pxt.create_dir`][pixeltable.create_dir]('experiments')              |
+| Rename or move a directory              | [`pxt.move`][pixeltable.move]('experiments', 'project_x.experiments') |
+| Delete a directory                      | [`pxt.drop_dir`][pixeltable.drop_dir]('experiments')                  |
+| Deleta a directory and all its contents | [`pxt.drop_dir`][pixeltable.drop_dir]('experiments', force=True)      |
```
Comment (mkornacker) [inline docs/release/api-cheat-sheet.md]: Delete

File: pixeltable/globals.py [hunk 2]
```diff
@@ -340,31 +346,48 @@ def rm_dir(path_str: str) -> None:
...
+                obj = cat.paths[child]
+            except excs.Error:
+                continue
+            if isinstance(obj, catalog.Dir):
+                drop_dir(str(child), force=True)
+            else:
+                assert isinstance(obj, catalog.Table)
+                assert not obj._is_dropped  # else it should have been removed from `cat.paths` already
+                drop_table(str(child), force=True)
+    elif len(children) > 0:
```
Comment (mkornacker) [inline pixeltable/globals.py]: Do this first in line 368 (!force and len(children) > 0) , then remove elif

**Review by mkornacker — APPROVED** [review]

**Review by aaron-siegel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/217)* status: 0 of 7 files reviewed, 2 unresolved discussions (waiting on @mkornacker)

___
*[`docs/release/api-cheat-sheet.md` line 36 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/217#gh-1681964739:-O22girMBhjpU22wfLW3:b-x8wb6n) ([raw file](https://github.com/pixeltable/pixeltable/blob/96df1c138c5d225645a59b34f299269f43cba355/docs/release/api-cheat-sheet.md#L36)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

Delete
</blockquote></details>

Oops thanks
___
*[`pixeltable/globals.py` line 383 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/217#gh-1681966644:-O22hHPfAy2aWD70aqT3:b32wtd) ([raw file](https://github.com/pixeltable/pixeltable/blob/96df1c138c5d225645a59b34f299269f43cba355/pixeltable/globals.py#L383)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

Do this first in line 368 (!force and len(children) > 0) , then remove elif
</blockquote></details>

ok

File: pixeltable/globals.py [hunk 3]
```diff
@@ -365,6 +365,10 @@ def drop_dir(path_str: str, force: bool = False, ignore_errors: bool = False) ->
             raise e
 
     children = cat.paths.get_children(path, child_type=None, recursive=True)
+
+    if len(children) > 0 and not force:
+        raise excs.Error(f'Directory `{path_str}` is not empty.')
+
     if force:
```
Comment (mkornacker) [inline pixeltable/globals.py:372]: that check is now superfluous (ie, you can remove line 72 and unindent the block)

**Review by mkornacker — APPROVED** [review]

**Pull request merged**